nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -169,9 +169,13 @@ class Costmap2DROS : public nav2_util::LifecycleNode
   /** @brief Same as getLayeredCostmap()->isCurrent(). */
   bool isCurrent()
   {
+    // lock the costmap because no costmap-reset is allowed until the isCurrent() finished
+    std::unique_lock<Costmap2D::mutex_t> lock(*(layered_costmap_->getCostmap()->getMutex()));
+    
     return layered_costmap_->isCurrent();
   }
 
+
```

Comment:
```Diff with 5 lines
--- include/nav2_costmap_2d/costmap_2d_ros.hpp
+++ include/nav2_costmap_2d/costmap_2d_ros.hpp.uncrustify
@@ -174 +174 @@
-    
+
```

Also, please pull in main or rebase, we need CI green and it should be green now